File: Shared/CPP/HCMTrack.cpp
//
//  HCMTrack.m
//  CPP
//
//  Copyright (c) 2015 Harry Cheung. All rights reserved.
//

#include "HCMTrack.h"

HCMTrack::HCMTrack(std::vector<HCMGate*> *gates, HCMGate *start) {
  this->gates = gates;
  this->start = start;
}

HCMTrack::~HCMTrack() {
  delete gates;
}

File: CPPAndroid/app/src/main/java/harrycheung/map/cppandroid/HCMTrack.java
//
// Copyright (c) 2015 Harry Cheung
//

package harrycheung.map.cppandroid;

public class HCMTrack {
  public long nativePointer;

  HCMTrack(long[] gates, long start) {
    nativePointer = loadTrack(gates, start);
  }

  private static native long loadTrack(long[] gates, long start);
}
